doc/compiler/optimizer/IntroReadLogFile.md

```diff
@@ -0,0 +1,1188 @@
+0x7fef13a746d0 ffffffd0 [0x7fef12ad0d80] 48 bf 78 2c 34 00 00 00 00 00      mov	rdi, 0x0000000000342c78	# MOV8RegImm64
+
+0x7fef13a74700 00000000 [0x7fef12d881e0] 48 8b 44 24 18                     mov	rax, qword ptr [rsp+0x18]		# L8RegMem, SymRef [#1164 +24]
+```
+
+`"Post Binary Instructions"` is the last thing in the JIT compilation.
+
+`0x7fef13a74700`: the code cache address or instruction address where the bytecode instruction is.
+
+`00000000`: instruction offset from start of method.
+
+`0x7fef12ad1d50`: TR instruction address.
+
+`48 8b 44 24 18`: bytes that represents the instruction.
+
+Code gen also generates prologue and epilogue. The linkage convention requires to save certain values on the way into
+a call, set up the stack pointer, grow the stack for a frame, check stack overflow etc.
+
+### 7.8 GC Maps
+
+```
+Internal stack atlas:
+  numberOfMaps=92
+  numberOfSlotsMapped=23
+  numberOfParmSlots=1
+  parmBaseOffset=24
+  localBaseOffset=-176
+
+  Locals information :
+  Local [0x7fef256df590] (GC map index :   0, Offset :  24, Size : 8) is an uninitialized collected parm
+  Local [0x7fef256df7c0] (GC map index :  -1, Offset :   8, Size : 8) is an uninitialized uncollected parm
+  Local [0x7fef12cac870] (GC map index :  17, Offset : -48, Size : 8) is an uninitialized uncollected auto
+  Local [0x7fef12cac630] (GC map index :  16, Offset : -56, Size : 8) is an uninitialized uncollected auto
+  Local [0x7fef12cac3f0] (GC map index :  20, Offset : -24, Size : 8) is an uninitialized uncollected auto
+…
+  Local [0x7fef262476b0] (GC map index :  -1, Offset : -288, Size : 4) is an uninitialized uncollected auto
+```
+GC maps are associated with instructions, or more correctly, ranges of instructions. When a GC occurs, the address of
+the current instruction (i.e. the address in the program counter register) is used to search for the appropriate map
+in the GC atlas of the method.
+
+The GC map describes all live autos/parms/registers that contain addresses at the given point in the program; these
+addresses are fixed up one by one by calling a GC routine that computes the displacement to apply for each address.
+
+We keep track of every point that could result in GC. Every local variable is searchable. All local variables have GC
+map index. The ones that are collected references (collected autos, collected register, and collected parms) have
+positive or non-negative GC map index.
+
+```
+  Map number : 2
+  Code offset range starts at [00001885]
+  GC stack map information :
+    number of stack slots mapped = 21
+    live stack slots containing addresses --> {0,5}
+  GC register map information :
+slot pushes: 0    registers: {eax }
+```
+Different GC map for different GC points. For GC `Map 2`, there are `21` collected autos. Out of the `21` autos, Stack
+slot index `0` and `5` are live at this point. Stack Walker knows where they are. Register `eax` holds collected
```

Comment:
> Stack Walker knows where they are.

During a stack walk, these indices are used mapped to the specific stack slots that will be checked for collected references.

Reply:
Addressed in 9c5fd3b